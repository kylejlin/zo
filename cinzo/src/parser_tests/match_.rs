use super::*;

#[test]
fn match_() {
    let src = r#"
(
    match

    0

    3

    // Cases
    (
        1

        2
    )
)"#;
    let tokens = lex(src).unwrap();
    let cst = parse(tokens).unwrap();
    insta::assert_debug_snapshot!(&cst);
}

#[test]
fn non_parenthesized_cases() {
    let src = r#"
(
    match

    0

    3

    1
)"#;
    let tokens = lex(src).unwrap();
    let err = parse(tokens).unwrap_err();
    insta::assert_debug_snapshot!(&err);
}
