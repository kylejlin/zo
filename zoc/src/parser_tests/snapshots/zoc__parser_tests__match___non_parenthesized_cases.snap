---
source: src/parser_tests/match_.rs
expression: "&err"
---
Some(
    Number(
        NumberLiteral {
            value: 1,
            span: (
                ByteIndex(
                    32,
                ),
                ByteIndex(
                    33,
                ),
            ),
        },
    ),
)
